component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordChangeEnforcerOnExpiration.java

```diff
@@ -233,7 +234,12 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                     log.debug("Updated user credentials of " + tenantAwareUsername);
                 }
             } catch (org.wso2.carbon.user.core.UserStoreException e) {
-                throw new AuthenticationFailedException("Incorrect current password", e);
+                if(e.getCause() instanceof PrivilegedActionException) {
+                    throw new AuthenticationFailedException(
+                            PasswordChangeEnforceConstants.PASSWORD_POLICY_VIOLATION_ERROR);
```

Comment:
1. Can you include something indicating the violation reason on the exception, since you not include e.
2. Need a debug log to put more details as to what caused this exception.